enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,260 @@
+title: single-node-deployment-with-bootstrap-in-place
+authors:
+  - "@dhellmann"
+  - "@eranco"
+  - "@romfreiman"
+  - "@tsorya"
+reviewers:
+  - TBD
+  - "@mrunalp"
+  - "@markmc"
+  - "@deads2k"
+  - "@wking"
+  - "@eparis"
+  - "@hexfusion"
+approvers:
+  - TBD
+creation-date: 2020-12-13
+last-updated: 2020-12-13
+status: implementable
+see-also:
+  - https://github.com/openshift/enhancements/pull/560
+  - https://github.com/openshift/enhancements/pull/302
+---
+
+# Single Node deployment with bootstrap-in-place
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+As we add the new [single-node production deployment](https://github.com/openshift/enhancements/pull/560/files)
+we need a way to install such cluster without an extra node dependency for bootstrap.
+
+This enhancement describes the flow for installing Single Node OpenShift using liveCD that perform the bootstrap logic and reboots to become the single node.
+
+## Motivation
+
+Currently, all OpenShift installations use an auxiliary bootstrap node.
+The bootstrap node creates a temporary control plane that is required for launching the actual cluster.
+
+Resource constraints are the primary motivation for single-node OCP clusters, and removing the need for a
+separate bootstrap machine significantly reduces the resources required to install the cluster.
+
+Requiring the auxiliary node for installing a Single Node OpenShift brings to play some requirements:
+1. The obvious additional node.
+2. Requires external dependencies:
+   a. Load balancer (only for bootstrap phase)
+   b. Requires DNS (configured per installation)
+3. Cannot use Bare Metal IPI:
+   a. Adds irrelevant dependencies - vips, keepalived, mdns
+   b. Requires same L2 between bootstrap and the Single Node OpenShift
```

Comment:
3.b. is no longer true.

Reply:
why not? If using baremeatl platform, then there is such limitation.
Using none platform, I agree, but then the user will have to provide external load balancer

Reply:
As of 4.7 it is possible to disable the provisioning network entirely when using virtual media. That's how we're going to use metal3 to attach the agent ISO to single-node hosts, and it could be used with IPI directly as well. https://github.com/openshift/cluster-baremetal-operator/blob/master/api/v1alpha1/provisioning_types.go#L96

Reply:
right.